src/headphones2/postprocess/extensions/renamer.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+from __future__ import (absolute_import, division,
+                        print_function, unicode_literals)
+import py
+import os
+import shutil
+
+import logbook
+
+from beetsplug.ftintitle import split_on_feat
+
+from headphones2.postprocess.component_base import PostProcessorComponentBase
+
+logger = logbook.Logger(__name__)
+Path = py.path.local
+
+class Renamer(PostProcessorComponentBase):
+
+    def __init__(self):
+        super(Renamer, self).__init__()
+
+    @staticmethod
+    def _components_from_item(item):
+        """
+        :param item:
+        :type item: beets.library.Item
+        :return: dictionary of track components
+        :rtype dict
+        """
+        components = {
+            "$Album": item.album,
+            "$Track_name": item.title,
+            "$Track_num": item.track,
+            "$Artist": item.artist,
+            "$SortArtist": split_on_feat(item.artist)[0],  # Gorillaz ft the doors --> ('Gorillaz', 'The Doors')
+            "$DiscNumber": item.disc,
+            "$Year": item.year,
+            "$Genre": item.genre
+        }
+
+        return {k: unicode(v) for k, v in components.iteritems()}
+
+    @staticmethod
+    def process(item_list, name_string="$SortArtist/$Album [$Year]/ $Track_num - $Track_name",
+                destination_folder=unicode(Path(os.path.expanduser("~")).join("Music")),
+                release_id=None, should_move=False):
+
+        # decide if we want to keep original files
+        operation = shutil.move if should_move else shutil.copy
+
+        for item in item_list:
+            components = Renamer._components_from_item(item)
+
+            original_path = Path(item.path)
+            new_name = name_string
+
+            for param, value in components.iteritems():
```

Comment:
There must be a better way! Why not use 'format()' and named places ('{artist} {album}...')